sdk/src/fee.rs

```diff
@@ -83,7 +96,7 @@ impl FeeStructure {
         _lamports_per_signature: u64,
         budget_limits: &FeeBudgetLimits,
         include_loaded_account_data_size_in_fee: bool,
-    ) -> u64 {
+    ) -> FeeDetails {
```

Comment:
this is changing the signature of a public interface, which is technically against semver. who's consuming this?

Reply:
direct caller (besides tests) are `bank` (calculate tx fee for distribution) and `accounts` (to validate payer account), both are updated in this PR. Don't think it is exposed to external 🤞🏼

Reply:
well it's _exposed_ for sure 'cause sdk is a publicly consumable crate and this method is publicly exported. question is whether it's actually _consumed_ by anyone, which we really ought not find out the hard way.

i suppose the most pedantic thing to do here would be to move this logic to a new method `calculate_fee_details(...) -> FeeDetails`, then deprecate `calculate_fee(...) -> u64` replacing its body with a call to `calculate_fee_details(...).total_fees()`

Reply:
I agree with Trent here, it's quite possible 3rd party code usees this function.
Though I'm not sure we need to actually deprecate `calculate_fee`, so long as the impl is just `calculate_fee_details(...).total_fees()`.

Seems reasonable there are many call-sites that don't care about the details, just the total fee. Might simplify this PR as well.

Reply:
Agree too. But does `calculate_fee()` need to be exposed to external? Users use simulator to get tx fee, are they? github says this function was moved from `runtime::bank` to `sdk::fee_structure` not too long ago. Maybe entire `fee_structure` belong to solana_runtime instead of sdk?

Reply:
the ship has sailed on whether it _needs_ to be or not. we've already shipped releases with it exposed in the public api of a publicly consumable crate that we've promised to follow semver on. removing that symbol for public availability anytime but a major release will be a violation of semver

Reply:
033c98b68eeb61f09d90c451739f80c9ab0a9738 for the sailed ship